File: CreationalPatterns/Prototype/CSharp/Rectangle.cs
namespace Prototype
{
    public class Rectangle : Shape
    {
        public int Width { get; set; }
        public int Height { get; set; }

        public Rectangle() { }
        public Rectangle(Rectangle source) : base(source)
        {
            Width = source.Width;
            Height = source.Height;
        }

        public override Rectangle Clone()
        {
            return new Rectangle(this);
        }

        public override string ToString()
        {
            return $"Rectangle: X={X}, Y={Y}, Color={Color}, Width={Width}, Height={Height}";
        }
    }
}

File: CreationalPatterns/Prototype/TypeScript/Rectangle.ts
import { Shape } from './Shape';

export class Rectangle extends Shape {
    width: number = 0;
    height: number = 0;

    constructor(source?: Rectangle) {
        super(source);
        if (source) {
            this.width = source.width;
            this.height = source.height;
        }
    }

    clone(): Rectangle {
        return new Rectangle(this);
    }

    toString(): string {
        return `Rectangle: x=${this.x}, y=${this.y}, color=${this.color}, width=${this.width}, height=${this.height}`;
    }
}

